pandas/core/series.py

```diff
@@ -3659,21 +3659,62 @@ def argsort(self, axis=0, kind="quicksort", order=None) -> Series:
         See Also
         --------
         numpy.ndarray.argsort : Returns the indices that would sort this array.
-        """
-        values = self._values
-        mask = isna(values)
 
-        if mask.any():
-            result = Series(-1, index=self.index, name=self.name, dtype="int64")
-            notmask = ~mask
-            result[notmask] = np.argsort(values[notmask], kind=kind)
-            return self._constructor(result, index=self.index).__finalize__(
+        Examples
+        --------
+
+        Argsorting a basic Series.
+
+        >>> series = Series([30, 10, 20], index=["high", "low", "mid"], name="xy")
+        >>> series.argsort()
+        low     1
+        mid     2
+        high    0
+        Name: xy, dtype: int64
+
+        Argsorting a Series with null values.
+
+        >>> series = Series([30, 10, np.nan, 20], name="xy",
+        ...                    index=["high", "low", "null", "mid"])
+        >>> series.argsort()
+        low     1
+        mid     3
+        high    0
+        null   -1
+        Name: xy, dtype: int64
+        """
+        values = self.values
+        na_mask = isna(values)
+        if not any(na_mask):
+            res = np.argsort(values, kind=kind)
+            res_ser = Series(res, index=self.index[res], dtype="int64", name=self.name)
+            return res_ser.__finalize__(self, method="argsort")
+        else:
+            # GH 42090
```

Comment:
would be very clear that we are ordering nulls at the `na_position='last'` (equiv of sort_values). i think its fine not to expose this as an option (as numpy doesn't).